ethereum/contracts/ERC20App.sol

```diff
@@ -73,37 +56,27 @@ contract Bank {
         return abi.encode(_sender, _recipient, _tokenAddr, _amount, _nonce);
     }
 
-    function submit(bytes calldata data)
-        external
-    {
-        // TODO: decode message bytes into (tag, recipient, amount) or (tag, recipient, token, amount)
-         uint256 tag = 0;                        // placeholder
-         address payable recipient = address(0); // placeholder
-         address token = address(0);             // placeholder
-         uint256 amount = 1;                     // placeholder
-
-        if(tag == uint(AppEventTags.SendETH)) {
-            unlockETH(recipient, amount);
-        } else if(tag == uint(AppEventTags.SendERC20)) {
-            unlockERC20(recipient, token, amount);
-        }
-    }
-
-    function unlockETH(
-        address payable _recipient,
-        uint256 _amount
-    )
-        internal
+    function unlockERC20(bytes memory _data)
```

Comment:
We need to add a signature field as per the milestone requirements. I'm submitting a signature along with the data in my relayer PR: https://github.com/Snowfork/polkadot-ethereum/pull/58/files#diff-6f08a58ca4132c1dcf83cb21d2575b5eR129

Reply:
Can we also rename the function to `submit` to match what the relayer would call?

Reply:
So the method signature should be `submit(bytes memory _data, bytes memory _signature)`. Thoughts?

Reply:
I've made this change in the integration branch at https://github.com/Snowfork/polkadot-ethereum/pull/59

Reply:
No strong opinion, but to me `submit` is much less clear than `unlock...`. Just my 2 cents ;)

Reply:
See https://github.com/Snowfork/polkadot-ethereum/pull/57#issuecomment-683683514